File: rest-service/src/main/java/mb/lib/blast/model/x/BlastxTask.java
package mb.lib.blast.model.x;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import mb.api.service.util.ErrorText;

public enum BlastxTask
{
  BlastX("blastx"),
  BlastXFast("blastx-fast");

  private static final BlastxTask def = BlastX;

  private final String value;

  BlastxTask(String value) {
    this.value = value;
  }

  @JsonValue
  public String getValue() {
    return value;
  }

  public static BlastxTask getDefault() {
    return def;
  }

  @JsonCreator
  public static BlastxTask fromString(String value) {
    for (var e : values())
      if (e.value.equals(value))
        return e;

    throw new IllegalArgumentException(String.format(
      ErrorText.InvalidEnumValue,
      value,
      BlastxTask.class.getSimpleName()
    ));
  }

  @Override
  public String toString() {
    return value;
  }
}

File: lib-jvm-mblast-sdk/src/main/kotlin/org/veupathdb/lib/mblast/sdk/query/blast/BlastXTask.kt
package org.veupathdb.lib.mblast.sdk.query.blast

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue
import com.fasterxml.jackson.databind.JsonNode
import com.fasterxml.jackson.databind.node.TextNode
import org.veupathdb.lib.mblast.sdk.util.UnrecognizedEnumException
import org.veupathdb.lib.mblast.sdk.util.dEnumRequireTextual

enum class BlastXTask {
  BlastX,
  BlastXFast,
  ;

  @JsonValue
  fun toJson(): JsonNode = TextNode(when (this) {
    BlastX     -> "blastx"
    BlastXFast -> "blastx-fast"
  })

  companion object {

    @JvmStatic
    @JsonCreator
    fun fromJson(js: JsonNode): BlastXTask {
      js.dEnumRequireTextual(BlastXTask::class)

      return when (val v = js.textValue()) {
        "blastx"      -> BlastX
        "blastx-fast" -> BlastXFast
        else          -> throw UnrecognizedEnumException(BlastXTask::class, v)
      }
    }
  }
}